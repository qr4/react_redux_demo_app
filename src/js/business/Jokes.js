// @flow

export class Jokes {
    +listOfJokes: string[];

    constructor(initialListOfJokes: string[] = []) {
        this.listOfJokes = initialListOfJokes;
    }

    addNewJoke(joke: string): Jokes {
        return new Jokes([...this.listOfJokes, joke]);
    }
}
