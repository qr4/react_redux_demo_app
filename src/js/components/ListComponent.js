// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ListGroup, ListGroupItem } from 'react-bootstrap';
// import type { Dispatch } from 'redux';

import type { State } from '../state/Reducer';
import { Jokes } from '../business/Jokes';

type Props = {
    jokes: Jokes,
};

export class DisconnectedListComponent extends Component<Props> {
    render() {
        return (
            <ListGroup>
                {this.props.jokes.listOfJokes.map((value, index) => (
                    <ListGroupItem key={'joke-' + index} className={'clearfix list-group-item-test-' + index}>
                        {value}
                    </ListGroupItem>
                ))}
            </ListGroup>
        );
    }
}

const mapStateToProps = (state: State): Props => ({ jokes: state.jokes });

export const ListComponent = connect(mapStateToProps)(DisconnectedListComponent);
