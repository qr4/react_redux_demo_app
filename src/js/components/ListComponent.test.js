// @flow

import React from 'react';
import { shallow } from 'enzyme';

import { DisconnectedListComponent } from './ListComponent';
import { Jokes } from '../business/Jokes';

describe('ListComponent', () => {
    function renderComponentSanely(props = {}) {
        const defaultProps = {
            jokes: new Jokes(),
        };

        return <DisconnectedListComponent {...defaultProps} {...props} />;
    }

    it('[snapshot] it renders correctly', () => {
        const app = shallow(renderComponentSanely());
        expect(app).toMatchSnapshot();
    });

    it('renders a list group item per joke', () => {
        const jokes = new Jokes().addNewJoke('this is a totally awesome joke!');
        const app = shallow(renderComponentSanely({ jokes: jokes }));

        const element = app.find('.list-group-item-test-0');
        expect(element.contains('this is a totally awesome joke!')).toBe(true);
    });
});
