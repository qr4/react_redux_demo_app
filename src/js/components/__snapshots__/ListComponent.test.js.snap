// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent [snapshot] it renders correctly 1`] = `
<ListGroup
  bsClass="list-group"
/>
`;
